### PR TITLE
feat!: expose organization apis of logging

### LINKED ISSUE #159: Expose organization APIs of logging
Implements the following [Organization ](https://cloud.google.com/logging/docs/reference/v2/rest/v2/organizations)APIs of logging.

-  [getcmeksettings](https://cloud.google.com/logging/docs/reference/v2/rest/v2/organizations#getcmeksettings) : Gets the Logs Router CMEK settings for the given resource.

- [updatecmeksettings](https://cloud.google.com/logging/docs/reference/v2/rest/v2/organizations#updatecmeksettings) : Updates the Logs Router CMEK settings for the given resource.

### PR DESCRIPTION
**Pull request by athakor**
Fixes #159

### CODE REVIEW

**Review by elharo — CHANGES_REQUESTED** [review]
Adding methods to interfaces is an incompatible change that requires a major version bump.

**Comment by chingor13** [comment]
@athakor We're ready to start prepping for the 2.0 version. Can you resolve the conflicts and repush?

**Comment by athakor** [comment]
> @athakor We're ready to start prepping for the 2.0 version. Can you resolve the conflicts and repush?

@chingor13 done PTAL

File: google-cloud-logging/clirr-ignored-differences.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- see http://mojo.codehaus.org/clirr-maven-plugin/examples/ignored-differences.html -->
+<differences>
+    <difference>
```
[the remark this answers is not in the document]
Comment (athakor) [reply google-cloud-logging/clirr-ignored-differences.xml]: done

**Comment by athakor** [comment]
@simonz130 PTAL when you get chance.

**Comment by athakor** [comment]
@simonz130 @chingor13  PTAL

File: google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java [hunk 2]
```diff
@@ -175,9 +178,10 @@
   ApiFuture<Empty> delete(DeleteLogMetricRequest request);
 
   /**
-   * Sends a request to create a new exclusion in a specified parent resource. This method returns
-   * an {@code ApiFuture} object to consume the result. {@link ApiFuture#get()} returns the created
-   * exclusion.
+<<<<<<< HEAD
+   * <<<<<<< HEAD Sends a request to create a new exclusion in a specified parent resource. This
+   * method returns an {@code ApiFuture} object to consume the result. {@link ApiFuture#get()}
+   * returns the created exclusion.
```
[the remark this answers is not in the document]
Comment (athakor) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java]: done

File: google-cloud-logging/src/main/java/com/google/cloud/logging/Logging.java [hunk 3]
```diff
@@ -669,8 +669,8 @@ public static EntryListOption filter(String filter) {
   ApiFuture<Boolean> deleteMetricAsync(String metric);
 
   /**
-   * Creates a new exclusion in a specified parent resource. Only log entries belonging to that
-   * resource can be excluded. You can have up to 10 exclusions in a resource.
+   * Creates a new exclusion in a specified parent resource. Only log entries belonging
+   * to that resource can be excluded. You can have up to 10 exclusions in a resource.
```
[the remark this answers is not in the document]
Comment (athakor) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/Logging.java]: done

File: google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java [hunk 4]
```diff
@@ -175,9 +178,9 @@
   ApiFuture<Empty> delete(DeleteLogMetricRequest request);
 
   /**
-   * Sends a request to create a new exclusion in a specified parent resource. This method returns
-   * an {@code ApiFuture} object to consume the result. {@link ApiFuture#get()} returns the created
-   * exclusion.
+   * Sends a request to create a new exclusion in a specified parent resource. This
+   * method returns an {@code ApiFuture} object to consume the result. {@link ApiFuture#get()}
+   * returns the created exclusion.
```
[the remark this answers is not in the document]
Comment (athakor) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java]: done

**Comment by athakor** [comment]
@simonz130 PTAL

**Review by simonz130 — APPROVED** [review]

**Comment by simonz130** [comment]
@elharo - we released v2 (breaking) and now I think this change can be merged. can you sigh this off?

File: google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java [hunk 5]
```diff
@@ -218,4 +221,22 @@
    * @param request the request object containing all of the parameters for the API call
    */
   ApiFuture<Empty> delete(DeleteExclusionRequest request);
+
+  /**
+   * Sends a request to get the CMEK settings. This method returns a {@code ApiFuture} object to
```
Comment (elharo) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java:226]: a --> an

File: google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java [hunk 6]
```diff
@@ -218,4 +221,22 @@
...
+   * consume the result.{@link ApiFuture#get()} returns the requested CMEK settings or {@code null}
+   * if the CMEK settings was not found.
+   *
+   * @param request
```
Comment (elharo) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java:230]: either document or remove

File: google-cloud-logging/clirr-ignored-differences.xml [hunk 7]
```diff
@@ -95,4 +95,39 @@
     <method>com.google.api.core.ApiFuture update(com.google.logging.v2.UpdateExclusionRequest)</method>
     <differenceType>7012</differenceType>
   </difference>
+  <difference>
```
Comment (elharo) [inline google-cloud-logging/clirr-ignored-differences.xml:98]: This requires a major version bump, to 3.0 I think.

File: google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java (hunk 5 above)
Comment (athakor) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java]: done

File: google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java (hunk 6 above)
Comment (athakor) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/LoggingRpc.java]: done

File: google-cloud-logging/clirr-ignored-differences.xml (hunk 7 above)
Comment (athakor) [reply google-cloud-logging/clirr-ignored-differences.xml:98]: > This requires a major version bump, to 3.0 I think.

cc: @simonz130

Comment (elharo) [reply google-cloud-logging/clirr-ignored-differences.xml:98]: I think the way to indicate a major version bump is to put an exclamation point after "feat" in the PR description. @BenWhitehead is that correct?

File: google-cloud-logging/clirr-ignored-differences.xml [hunk 8]
```diff
@@ -95,4 +95,39 @@
...
+    <differenceType>7012</differenceType>
+  </difference>
+  <difference>
+    <className>com/google/cloud/logging/spi/v2/LoggingRpc</className>
+    <method>com.google.api.core.ApiFuture updateCmekSettingsAsync(com.google.cloud.logging.CmekSettings)</method>
+    <differenceType>7012</differenceType>
+  </difference>
+  <difference>
+    <className>com/google/cloud/logging/spi/v2/LoggingRpc</className>
+    <method> com.google.api.core.ApiFuture getCmekSettings(com.google.logging.v2.GetCmekSettingsRequest)</method>
```
Comment (elharo) [inline google-cloud-logging/clirr-ignored-differences.xml:125]: You might need to remove the space after `<method>` here. I'm not sure.

Comment (athakor) [reply google-cloud-logging/clirr-ignored-differences.xml]: done

**Comment by losalex** [comment]
@athakor , I wonder if this change still relevant - if not, we probably should close this PR

**Pull request closed without merging**